source/_docs/installation/synology.markdown

```diff
@@ -2,78 +2,177 @@
 
-
-The following configuration has been tested on Synology 413j running DSM 6.0-7321 Update 1.
+The following configuration has been tested on [Synology DS115j](https://www.synology.com/en-global/products/DS115j) running DSM 6.2.1-23824 Update 1.
 
 Running these commands will:
+* Compile the required new Python 3 package
+* Install Home Assistant
+* Enable Home Assistant to be launched on http://*Synology_IP*:8123
 
- - Install Home Assistant
- - Enable Home Assistant to be launched on [http://localhost:8123](http://localhost:8123)
-
-Using the Synology webadmin:
-
- - Install python3 using the Synology Package Center
- - Create homeassistant user and add to the "users" group
+Using the [Synology webadmin](https://www.synology.com/en-global/knowledgebase/DSM/help):
+* Install compiled Python 3 package using the [Synology Package Center](https://www.synology.com/en-global/knowledgebase/DSM/help/DSM/PkgManApp/PackageCenter_desc)
+* Create "homeassistant" user
+* Create "homeassistant" [Shared-Folder](https://www.synology.com/en-global/knowledgebase/DSM/help/DSM/AdminCenter/file_share_desc)
+* Enable [SSH](https://www.synology.com/en-global/knowledgebase/DSM/help/DSM/AdminCenter/system_terminal) service
+* Optionally start Home Assistant on bootup of your Synology NAS
 
-SSH onto your synology & login as admin or root
+## {% linkable_title Making new Python 3 package %}
 
- - Log in with your own administrator account
- - Switch to root using:
+#### {% linkable_title Installing Docker on [Mint, Ubuntu or derivatives](https://docs.docker.com/install/linux/docker-ce/ubuntu/) %}
 
+Install Docker:
 ```bash
-$ sudo -i
+# sudo apt-get update
+# sudo apt-get install docker.io
+```
+Start Docker and enable start on system bootup:
+```bash
+# sudo systemctl start docker
+# sudo systemctl enable docker
 ```
 
+#### {% linkable_title Preparing compiling environment %}
 
-Check the path to python3 (assumed to be /volume1/@appstore/py3k/usr/local/bin)
-
+Install Git:
+```bash
+# sudo apt-get install git
+```
+Fork and clone [spksrc](https://github.com/SynoCommunity/spksrc), this will make a directory "~/spksrc":
 ```bash
-# cd /volume1/@appstore/py3k/usr/local/bin
+git clone https://github.com/SynoCommunity/spksrc.git ~/spksrc
 ```
 
-Install PIP (Python's package management system)
+Edit the following 2 files.
 
-```bash
-# ./python3 -m ensurepip
+Add these edits to make cross compiled Python module packages which Home Assistant needs to install and enable usage of the "[Cloud](/components/cloud/)" and [Homekit](/components/homekit/)" components. Most components will setup just fine, but certain components need a python package that compiles on installation, which will fail as Synology did not provide a compiler, causing said need for cross compilation.<br/>
```

Comment:
enable usage of the [Cloud](/components/cloud/) and [Homekit](/components/homekit/) components